### PR TITLE
feat: clamp to consensus max gas

### PR DESCRIPTION
**Pull request by aljo242**
Closes BLO-590

Adds the `consensusparams` keeper to the `fee` posthandler.  Whenever `consensus.MaxGas` < `feeMarket.MaxGas` the `consensus` value is used.  This effectively clamps our max gas to this value whenever we exceed it, but still allows the values to differ if `consensus.MaxGas` > `feeMarket.MaxGas`.

Also removes the `crisis` module from the app.  This module isnt needed for testing and contains vulnerabilities

## Summary by CodeRabbit

- **New Features**
  - Implemented new consensus parameter handling in fee market logic.
  - Enhanced fee deduction decorator to utilize updated consensus parameters.

- **Bug Fixes**
  - Fixed fee market state update logic to correctly use block utilization parameters.

- **Refactor**
  - Removed crisis module dependencies across tests and application configurations.
  - Introduced new `ConsensusKeeper` for managing consensus parameters.

- **Tests**
  - Updated test suites and mocks to align with new consensus parameter usage.
  - Added error handling and boundary case simulations for consensus parameters.

- **Documentation**
  - Updated comments to clarify units of measurement for block utilization.

- **Chores**
  - Cleaned up imports and unused references pertaining to the crisis module.

### CODE REVIEW

File: tests/app/app.go [hunk 1]
```diff
@@ -132,7 +129,6 @@ type TestApp struct {
 	MintKeeper            mintkeeper.Keeper
 	DistrKeeper           distrkeeper.Keeper
 	GovKeeper             *govkeeper.Keeper
-	CrisisKeeper          *crisiskeeper.Keeper
```
Comment (nivasan1) [inline tests/app/app.go:135]: q: Why remove?

File: tests/app/post.go [hunk 2]
```diff
@@ -30,12 +31,17 @@ func NewPostHandler(options PostHandlerOptions) (sdk.PostHandler, error) {
 		return nil, errorsmod.Wrap(sdkerrors.ErrLogic, "feemarket keeper is required for post builder")
 	}
 
+	if options.ConsensusParamsKeeper == nil {
```
Comment (nivasan1) [inline tests/app/post.go:34]: Were we not able to get `ConsensusParams` from the sdk.Context?

File: x/feemarket/post/fee.go [hunk 3]
```diff
@@ -99,9 +101,22 @@ func (dfd FeeMarketDeductDecorator) PostHandle(ctx sdk.Context, tx sdk.Tx, simul
...
 
-	err = state.Update(gas, params)
+	consensusParams, err := dfd.consensusParamsKeeper.Get(ctx)
 	if err != nil {
-		return ctx, errorsmod.Wrapf(err, "unable to update fee market state")
+		return ctx, sdkerrors.ErrKeyNotFound.Wrap(err.Error())
+	}
+
+	maxGas := params.MaxBlockUtilization
+	if consensusParams.Block.MaxGas < int64(params.MaxBlockUtilization) && consensusParams.Block.MaxGas >= 0 {
```
Comment (nivasan1) [inline x/feemarket/post/fee.go:110]: This can happen if the fee-market params are updated separately from the ConsensusParams?

Comment (nivasan1) [reply x/feemarket/post/fee.go:110]: Any reason why we wouldn't update `MaxBlockUtilization` (not saying we shld, j asking)

File: tests/app/app.go (hunk 1 above)
Comment (aljo242) [reply tests/app/app.go:135]: introduces a vulnerability that `govulncheck` complains about.  Not needed anyways since its just in the test app

File: tests/app/post.go (hunk 2 above)
Comment (aljo242) [reply tests/app/post.go:34]: This is a good point.  Would be a big simplification.  Idk if there is a difference between the values.  from the context and the keeper

Comment (nivasan1) [reply tests/app/post.go:34]: See `getContextForTx` in `baseapp.go`, TLDR, it reads the value from state on tx execution. One edge-case tho would be if `ConsensusParams` are modified in the tx that we are calculating fees for tho?

File: x/feemarket/post/fee.go (hunk 3 above)
Comment (nivasan1) [reply x/feemarket/post/fee.go:110]: Good point, that makes sense

File: tests/app/post.go (hunk 2 above)
Comment (nivasan1) [reply tests/app/post.go:34]: Ya looking at this now, the only way to update is from comet or from a gov-proposal (which would be executed in Begin/EndBlockers), so this shld be safu

Comment (nivasan1) [reply tests/app/post.go:34]: I'd prefer to use ctx.ConsensusParams, but if that's a big lift I'm down to backlog and keep things as they are now, LMK what is preferred

Comment (aljo242) [reply tests/app/post.go:34]: Agreed. Will be cleaner and simpler. I can make the change now!

**Review by Eric-Warehime — APPROVED** [review]
LGTM

**Comment by aljo242** [comment]
Closing as this is outdated and breaking
